src/jobs/PkgEvalJob.jl

```diff
@@ -108,13 +108,16 @@ function PkgEvalJob(submission::JobSubmission)
     end
 
     if haskey(submission.kwargs, :compiled)
-        val = Meta.parse(submission.kwargs[:compiled])
-        if !isa(val, Bool)
-            error("invalid argument to `compiled` keyword")
+        expr = Meta.parse(submission.kwargs[:compiled])
+        if !isa(expr, QuoteNode)
+            error("invalid argument to `compiled` keyword (should be a Symbol)")
+        end
+        compiled = expr.value
+        if !in(compiled, [:none, :primary, :against, :both])
```

Comment:
Could be nice to have a comment that explains the difference between these four options.

Reply:
There's an according change in the README. Or do you mean details on the values (which I assumed to be fairly self-explanatory, but I can add some more text).

Reply:
That comparison does take into account build flags and compilation mode: https://github.com/JuliaCI/Nanosoldier.jl/blob/eff19637c464e02af520f7eef6e181418472cc1a/src/jobs/PkgEvalJob.jl#L368-L370

So if any of those differ, an identical SHA will be accepted and a comparison build performed. So it's just `runtests(..., vs="...", compiled=:primary)`

Reply:
So, suppose that I comment the following on a specific commit:

```
@nanosoldier runtests(ALL, vs="FOO", compiled=:primary)
```

What do I put in place of `FOO`, to tell Nanosoldier that both the primary and the against should use the same commit hash (i.e. the commit that I post the comment on)?

Reply:
There isn't a special option for using the same. Just specify the commit or the branch.